### PR TITLE
fixes #12914 - update configurations to rails 4.1 structure

### PR DESCRIPTION
**Pull request by ohadlevy**
this is a simple output after comparing the output of `rails new foreman`
and updating accordingly.

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -1,164 +1,86 @@
...
-  end
-
-  # Serve plugin static assets if the application is configured to do so
-  if config.serve_static_assets
-    ::Rails::Engine.subclasses.map(&:instance).each do |engine|
-      if File.exist?("#{engine.root}/public/assets")
-        app.middleware.use ::ActionDispatch::Static, "#{engine.root}/public"
-      end
-    end
-  end
```
Comment (ehelms) [inline config/environments/production.rb:164]: Lines 127-163 are required for plugin asset support

File: config/environments/production.rb [hunk 2]
```diff
@@ -1,164 +1,86 @@
...
-                  class_edit
-                  dashboard
-                  auth_source_ldap
-                  subnets
-                  hidden_values
-                  password_strength)
-
-  stylesheets = %w( )
-
-  config.assets.precompile += javascript.map{|js| js + ".js"} + stylesheets + %w(background-size.htc)
```
Comment (ehelms) [inline config/environments/production.rb:123]: Pretty sure without lines 70-122, assets won't get precompiled properly.

File: config/environments/production.rb (hunk 1 above)
Comment (ehelms) [inline config/environments/production.rb:164]: Oh, I see you moved them to an initializer down below

File: config/environments/production.rb (hunk 2 above)
Comment (ohadlevy) [inline config/environments/production.rb:123]: they are moved to a separate file as rails by default now uses an assets initializer.

**Comment by ohadlevy** [comment]
rebased

**Comment by dLobatog** [comment]
Merged as 418e709df58c4114ddf77fc88aed8da31f1e63ff, thanks @ohadlevy!

**Pull request closed without merging**